programs/sunrise_beam/src/state.rs

```diff
@@ -70,93 +83,82 @@ impl ControllerState {
         if let Some(bump) = input.new_gsol_mint_authority_bump {
             self.gsol_mint_authority_bump = bump;
         }
+        if let Some(window) = input.new_alloc_window {
+            self.alloc_window = window;
+        }
         if let Some(yield_account) = input.new_yield_account {
             self.yield_account = yield_account;
         }
     }
 
     /// A return value of [None] indicates no space was found.
```

Comment:
I think we should be more explicit with this. This seems like an error case rather than a sensible use of Optional. It can be detcted by the client and the client can resize before calling this function, that way we keep each individual instruction simple and reduce unexpected side-effects (e.g. rent being paid sometimes and not other times)